lhotse/cut.py

```diff
@@ -448,6 +448,16 @@ def map_supervisions(self, transform_fn: Callable[[SupervisionSegment], Supervis
         new_cut = fastcopy(self, supervisions=[s.map(transform_fn) for s in self.supervisions])
         return new_cut
 
+    def filter_supervisions(self, supervision_ids: Iterable[str]) -> AnyCut:
```

Comment:
I was thinking of a signature that more closely corresponds to the one of Python's `filter` function, i.e. `filter_supervisions(self, predicate: Callable[[SupervisionSegment], bool] -> Cut`

It's what we did with `map_supervisions` too. It allows to use an arbitrary filtering criterion. Examples of use:
```python
cut = cut.filter_supervisions(lambda s: s.id in supervision_ids)
cut = cut.filter_supervisions(lambda s: s.duration < 5.0)
cut = cut.filter_supervisions(lambda s: s.text is not None)
```

Reply:
Good idea, lets also make it for the `CutSet.filter_supervisions(predicate)` instead of `CutSet.subset(supervision_ids)`. 
Getting a train subset based on ids: `CutSet.filter_supervisions(lambda s: s.id in trainset_ids)`. WYT?

Reply:
I am not sure TBH. I think we could add it, but the semantics are a bit different than those of `subset` - `filter_supervisions` only removes the supervisions, while `subset` would also remove the cuts. So we could have both actually. But I don't think that `subset` should accept `predicate` because we might want to add options like `n_shortest`, `n_longest`, etc. that are not easily expressed with a predicate.

Reply:
Um why would we want to keep cuts without supervisions if we remove all their supervisions by `.filter_supervisions`? It seems a bit cleaner if only one filtering option for supervisions exists. Having options such as `n_longest` would be useful though.  I pushed some updates - there is only `CutSet.filter_supervisions` right now, but we can easily revert back once the interface is clear. 

Is there some guideline on how to write tests for Lhotse? I adjusted the test for `CutSet.filter_supervisions` and added test for `Cut.filter_supervisions` Shoud they be adjusted somehow or moved somewhere else?

Reply:
In general a Cut without a supervision is a valid object in Lhotse, and it could be used for some kind of unsupervised learning, or as "in-domain" noise to be added to other cuts, etc. We might want to filter out some supervisions (e.g. too short, not containing text but e.g. indicating music/noise, etc.) and split the resulting CutSet into a supervised and unsupervised part, etc. 

I'm thinking of `filter_supervisions` as a "low/mid level" API (or a building block) that would likely be followed by other operations to make the CutSet actually useful, and of `subset` as a more "high level" API.

Reply:
I haven't written down a guideline for tests yet but it seems like a good idea, I'll add it to my todo-list. I will review the PR later/tomorrow and let you know.

Reply:
Ok to sum up:

```
CutSet.subset(subset_ids: Iterable[str] = None, ...space for future args...)  # removes cuts with empty supervisions
CutSet.filter_supervisions(predicate)   # only removes supervisions, can contain unsupervised cuts
Cut.filter_supervisions(predicate)  # removes supervisions that do not satisfy predicate, may return Cut without supervision
``` 
:)